libs/tokens/src/state/tokens/favoriteTokensAtom.ts

```diff
@@ -44,3 +45,40 @@ export const toggleFavoriteTokenAtom = atom(null, (get, set, token: TokenWithLog
     [chainId]: state,
   })
 })
+
+function migrateFavoriteTokensAtom(oldStorageKey: string, newStorageKey: string): void {
+  try {
+    const favoriteV1Raw = localStorage.getItem(oldStorageKey)
+
+    if (!favoriteV1Raw) {
+      return
+    }
+
+    const state = JSON.parse(favoriteV1Raw) as FavoriteTokens
+
+    // Replace USDC with USDC.e on Gnosis chain
+    state[SupportedChainId.GNOSIS_CHAIN] = Object.keys(state[SupportedChainId.GNOSIS_CHAIN]).reduce<TokensMap>(
+      (acc, address) => {
+        if (address.toLowerCase() === USDC_GNOSIS_CHAIN.address.toLowerCase()) {
```

Comment:
nit, save `USDC_GNOSIS_CHAIN.address.toLowerCase()` in a const to not do it in each iteration